include/nbl/builtin/hlsl/ieee754.hlsl

```diff
@@ -0,0 +1,128 @@
+		}
+
+		template<>
+		float64_t replaceBiasedExponent(float64_t x, uint32_t biasedExp)
+		{
+			return bit_cast<float64_t, uint64_t>(replaceBiasedExponent<uint64_t>(bit_cast<uint64_t, float64_t>(x), biasedExp));
+		}
+	}
+
+	template <typename T>
+	int getExponentBitCnt() { return 0xdeadbeefu; }
+	template<> int getExponentBitCnt<float16_t>() { return 5; }
+	template<> int getExponentBitCnt<uint16_t>() { return 5; }
+	template<> int getExponentBitCnt<float32_t>() { return 8; }
+	template<> int getExponentBitCnt<uint32_t>() { return 8; }
+	template<> int getExponentBitCnt<float64_t>() { return 11; }
+	template<> int getExponentBitCnt<uint64_t>() { return 11; }
+
+	template <typename T>
+	int getMantissaBitCnt() { return 0xdeadbeefu; }
+	template<> int getMantissaBitCnt<float16_t>() { return 10; }
+	template<> int getMantissaBitCnt<uint16_t>() { return 10; }
+	template<> int getMantissaBitCnt<float32_t>() { return 23; }
+	template<> int getMantissaBitCnt<uint32_t>() { return 23; }
+	template<> int getMantissaBitCnt<float64_t>() { return 52; }
+	template<> int getMantissaBitCnt<uint64_t>() { return 52; }
+
+	template <typename T>
+	int getExponentBias() { return 0xdeadbeefu; }
+	template<> int getExponentBias<float16_t>() { return 15; }
+	template<> int getExponentBias<uint16_t>() { return 15; }
+	template<> int getExponentBias<float>() { return 127; }
+	template<> int getExponentBias<uint32_t>() { return 127; }
+	template<> int getExponentBias<double>() { return 1023; }
+	template<> int getExponentBias<uint64_t>() { return 1023; }
+
+	template <typename T>
+	unsigned_integer_of_size<sizeof(T)>::type getExponentMask() { return 0xdeadbeefu; }
+	template<> unsigned_integer_of_size<2>::type getExponentMask<float16_t>() { return 0x7C00; }
+	template<> unsigned_integer_of_size<2>::type getExponentMask<uint16_t>() { return 0x7C00; }
+	template<> unsigned_integer_of_size<4>::type getExponentMask<float32_t>() { return 0x7F800000u; }
+	template<> unsigned_integer_of_size<4>::type getExponentMask<uint32_t>() { return 0x7F800000u; }
+	template<> unsigned_integer_of_size<8>::type getExponentMask<float64_t>() { return 0x7FF0000000000000ull; }
+	template<> unsigned_integer_of_size<8>::type getExponentMask<uint64_t>() { return 0x7FF0000000000000ull; }
+
+	template <typename T>
+	unsigned_integer_of_size<sizeof(T)>::type getMantissaMask() { return 0xdeadbeefu; }
+	template<> unsigned_integer_of_size<2>::type getMantissaMask<float16_t>() { return 0x03FF; }
+	template<> unsigned_integer_of_size<2>::type getMantissaMask<uint16_t>() { return 0x03FF; }
+	template<> unsigned_integer_of_size<4>::type getMantissaMask<float32_t>() { return 0x007FFFFFu; }
+	template<> unsigned_integer_of_size<4>::type getMantissaMask<uint32_t>() { return 0x007FFFFFu; }
+	template<> unsigned_integer_of_size<8>::type getMantissaMask<float64_t>() { return 0x000FFFFFFFFFFFFFull; }
+	template<> unsigned_integer_of_size<8>::type getMantissaMask<uint64_t>() { return 0x000FFFFFFFFFFFFFull; }
```

Comment:
I'd rather not have them as magic numbers, see
https://github.com/Devsh-Graphics-Programming/Nabla/pull/696/files#r1639449866

Reply:
see what

Reply:
the link leads nowhere, but i assume you talk about the `traits` struct

Reply:
done

Reply:
again don't resolve yourself

Btw I think this ways about the fact that you can compute the mantissa mask by knowing how many bits are set side for it